### PR TITLE
Transfer FIO Domain feature.

### PR DESCRIPTION
**Pull request by andreyvEze**

### CODE REVIEW

File: src/components/themed/AddressTile.js [hunk 1]
```diff
@@ -0,0 +1,259 @@
...
+          return showError(e.message)
+        }
+      }
+    }
+    try {
+      const parsedUri: EdgeParsedUri = await coreWallet.parseUri(address, currencyCode)
+
+      this.setState({ resolvingAddress: false })
+
+      if (parsedUri.token) {
```
Comment (andreyvEze) [inline src/components/themed/AddressTile.js:73]: remove

File: src/components/themed/AddressTile.js [hunk 2]
```diff
@@ -0,0 +1,259 @@
...
+        // TOKEN URI
+        // todo:
+        return showError(s.strings.scan_invalid_address_error_title)
+      }
+
+      if (isLegacyAddressUri(parsedUri)) {
```
Comment (andreyvEze) [inline src/components/themed/AddressTile.js:79]: leave, implement proper logic

File: src/components/themed/AddressTile.js [hunk 3]
```diff
@@ -0,0 +1,259 @@
...
+        // LEGACY ADDRESS URI
+        // todo: showLegacyAddressModal
+        return showError(s.strings.legacy_address_modal_title)
+      }
+      if (isPrivateKeyUri(parsedUri)) {
```
Comment (andreyvEze) [inline src/components/themed/AddressTile.js:84]: remove

File: src/components/themed/AddressTile.js [hunk 4]
```diff
@@ -0,0 +1,259 @@
...
+        // PRIVATE KEY URI
+        // todo:
+        return showError(s.strings.scan_invalid_address_error_title)
+      }
+      if (isPaymentProtocolUri(parsedUri)) {
```
Comment (andreyvEze) [inline src/components/themed/AddressTile.js:89]: leave, implement proper logic

File: src/components/modals/ScanModal.js [hunk 5]
```diff
@@ -0,0 +1,206 @@
...
+          </TouchableHighlight>
+        </View>
+      )
+    }
+
+    if (this.props.cameraPermission === 'authorized') {
+      const flashMode = this.props.torchEnabled ? RNCamera.Constants.FlashMode.torch : RNCamera.Constants.FlashMode.off
+
+      return (
+        <RNCamera style={styles.cameraArea} captureAudio={false} flashMode={flashMode} onBarCodeRead={this.onBarCodeRead} type={RNCamera.Constants.Type.back}>
```
Comment (andreyvEze) [inline src/components/modals/ScanModal.js:80]: Put in <View> with horizontal paddings.

File: src/components/modals/ScanModal.js [hunk 6]
```diff
@@ -0,0 +1,206 @@
...
+  close = () => {
+    this.props.bridge.resolve()
+  }
+
+  render() {
+    const { bridge, theme, title } = this.props
+    const styles = getStyles(theme)
+    return (
+      <AirshipModal bridge={bridge} onCancel={this.close} borderRadius={theme.rem(1)} backgroundColor={theme.tileBackground}>
+        <T style={styles.header}>{title}</T>
```
Comment (andreyvEze) [inline src/components/modals/ScanModal.js:107]: use ModalTitle

File: src/components/modals/ScanModal.js [hunk 7]
```diff
@@ -0,0 +1,206 @@
...
+        {this.renderCameraArea()}
+        <TouchableHighlight style={styles.closeButton} onPress={this.close} underlayColor={theme.secondaryButton}>
+          <Ionicon style={styles.flashIcon} name="ios-arrow-down" size={theme.rem(1.5)} />
```
Comment (andreyvEze) [inline src/components/modals/ScanModal.js:110]: use ModalCloseArrow

File: src/components/modals/ScanModal.js [hunk 8]
```diff
@@ -0,0 +1,206 @@
...
+  }
+
+  render() {
+    const { bridge, theme, title } = this.props
+    const styles = getStyles(theme)
+    return (
+      <AirshipModal bridge={bridge} onCancel={this.close} borderRadius={theme.rem(1)} backgroundColor={theme.tileBackground}>
+        <T style={styles.header}>{title}</T>
+        {this.renderCameraArea()}
+        <TouchableHighlight style={styles.closeButton} onPress={this.close} underlayColor={theme.secondaryButton}>
```
Comment (andreyvEze) [inline src/components/modals/ScanModal.js:109]: use ModalCloseArrow

File: src/components/modals/ScanModal.js [hunk 9]
```diff
@@ -0,0 +1,206 @@
...
+  }),
+  (dispatch: Dispatch) => ({
+    toggleEnableTorch: () => dispatch({ type: 'TOGGLE_ENABLE_TORCH' }),
+    disableScan: () => {
+      dispatch({ type: 'DISABLE_SCAN' })
+    },
+    enableScan: () => {
+      dispatch({ type: 'ENABLE_SCAN' })
+    },
+    requestPermission: (permission: Permission) => {
```
Comment (andreyvEze) [inline src/components/modals/ScanModal.js:202]: write just `requestPermission `

File: src/components/scenes/FioDomainSettingsScene.js [hunk 10]
```diff
@@ -99,6 +108,25 @@ export class FioDomainSettingsScene extends React.Component<Props, State> {
     this.setFee()
   }
 
+  onTransferPress = async () => {
+    const { fioWallet } = this.props
+    let amount
```
Comment (andreyvEze) [inline src/components/scenes/FioDomainSettingsScene.js:113]: move to `try`, make const

File: src/components/scenes/SendScene.js [hunk 11]
```diff
@@ -0,0 +1,314 @@
...
+type Props = OwnProps & StateProps & DispatchProps & ThemeProps
+
+type State = {
+  recipientAddress: string,
+  clipboard: string,
+  loading: boolean,
+  showSlider: boolean
+}
+
+class SendComponent extends React.Component<Props, State> {
```
Comment (andreyvEze) [inline src/components/scenes/SendScene.js:72]: PureComponent

File: src/components/scenes/SendScene.js [hunk 12]
```diff
@@ -0,0 +1,314 @@
...
+import { ScrollView, View } from 'react-native'
+import { Actions } from 'react-native-router-flux'
+import { connect } from 'react-redux'
+import { sprintf } from 'sprintf-js'
+
+import { FIO_ADDRESS_LIST } from '../../constants/SceneKeys'
+import { FIO_STR } from '../../constants/WalletAndCurrencyConstants'
+import s from '../../locales/strings.js'
+import { getPlugins, getSettings } from '../../modules/Settings/selectors.js'
+import FormattedText from '../../modules/UI/components/FormattedText/FormattedText.ui.js'
```
Comment (andreyvEze) [inline src/components/scenes/SendScene.js:16]: use `themed/EdgeText`

File: src/components/themed/AddressTile.js [hunk 13]
```diff
@@ -33,6 +57,26 @@ type State = {
 }
 type Props = OwnProps & StateProps & ThemeProps
 
+export const isLegacyAddressUri = (parsedUri: EdgeParsedUri): boolean => {
```
Comment (andreyvEze) [inline src/components/themed/AddressTile.js:60]: remove exports

**Review by thehobbit85 — APPROVED** [review]

**Pull request merged**